src/streaming-client.ts

```diff
@@ -0,0 +1,157 @@
+import * as events from 'events';
+import * as fs from 'fs';
+import { Duplex, PassThrough } from 'stream';
+import * as sockets from 'websocket';
+
+import { AudioConfig } from './models/streaming/AudioConfig';
+import { BufferedDuplex } from './models/streaming/BufferedDuplex';
+import {
+    StreamingConnected,
+    StreamingHypothesis,
+    StreamingResponse
+} from './models/streaming/StreamingResponses';
+
+/**
+ * Client which handles a streaming connection to the Rev.ai API.
+ * @event httpResponse emitted when the client fails to start a websocket connection and
+ *      receives an http response. Event contains the http status code of the response.
+ * @event connectFailed emitted when the client fails to begin a websocket connection and
+ *      received a websocket error. Event contains the received error.
+ * @event connect emitted when the client receives a connected message from the API. Contains
+ *      the StreamingConnected returned from the API.
+ * @event close emitted when the connection is properly closed by the server. Contains the
+ *      close code and reason.
+ * @event error emitted when an error occurs in the connection to the server. Contains the
+ *      thrown error.
+ *
+ * @example
+ * const audioConfig = new AudioConfig(<YOUR-CONTENT-TYPE-PARAMS>);
+ * const token = <YOUR-ACCESS-TOKEN>;
+ * var client = new RevAiStreamingClient(token, audioConfig);
+ * client.on('close', (code, reason) => {
+ *     console.log(`Connection closed, ${code}: ${reason}`);
+ * });
+ * client.on('httpResponse', code => {
+ *     console.log(code);
+ * })
+ * client.on('connectFailed', error => {
+ *     console.log(error);
+ * })
+ * client.on('connect', connectionMessage => {
+ *     console.log(connectionMessage);
+ * })
+ *
+ * var stream = client.start();
+ *
+ * var file = fs.createReadStream(<YOUR-AUDIO-FILE>);
+ * stream.on('data', data => {
+ *     console.log(data);
+ * });
+ * stream.on('end', function () {
+ *     console.log("End of Stream");
+ * });
+ *
+ * file.pipe(stream);
+ */
+export class RevAiStreamingClient extends events.EventEmitter {
+    baseUrl: string;
+    client: sockets.WebSocketClient;
+    private accessToken: string;
+    private config: AudioConfig;
+    private requests: PassThrough;
+    private responses: PassThrough;
+
+    /**
+     * @param accessToken Access token associated with the user's account
+     * @param config Configuration of the audio the user will send from this client
+     * @param version (optional) Version of the Rev.ai API the user wants to use
+     */
+    constructor(accessToken: string, config: AudioConfig, version = 'v1alpha') {
+        super();
+        this.accessToken = accessToken;
+        this.config = config;
+        this.baseUrl = `wss://api.rev.ai/speechtotext/${version}/stream?` +
```

Comment:
This same question applies to both async and streaming. I think we should make baseUrl configurable so it can be pointed at other environments. Default environment would be the production environment.
Was there prior discussion on this?

Reply:
Yes there was. We don't expose it as publicly configurable since that makes no sense to customers. But I leave it as not private so that we can assign it if we want to. 

ie we can do
`let client = new RevAiStreamingClient()`
`client.baseUrl = 'some other environment'`

Reply:
I guess it's a bit harder with this one as the access token and such are baked into the url. But I still think this is the best solution